File: 3309-count-prefix-and-suffix-pairs-i/3309-count-prefix-and-suffix-pairs-i.cpp
class Solution {
public:
    int countPrefixSuffixPairs(vector<string>& words) {
         int n = words.size();
        int ans = 0;

        auto isPrefixAndSuffix = [](const std::string& a, const std::string& b) -> bool {
            return b.compare(0, a.size(), a) == 0 && b.compare(b.size() - a.size(), a.size(), a) == 0;
        };

        for (int i = 0; i < n; ++i) {
            for (int j = i + 1; j < n; ++j) {
                if (isPrefixAndSuffix(words[i], words[j])) {
                    ans++;
                }
            }
        }
        return ans;
    }
};

File: 3309-count-prefix-and-suffix-pairs-i/3309-count-prefix-and-suffix-pairs-i.py
class Solution:
    def countPrefixSuffixPairs(self, words: List[str]) -> int:
        n=len(words)
        ans=0
        def isPrefixAndSuffix(a,b):
            return b.startswith(a) and b.endswith(a)
        for i in range(n):
            for j in range(i+1,n):
                if isPrefixAndSuffix(words[i],words[j]):
                    ans+=1
        return ans
        